csharp/src/Apache.Arrow.Adbc/Extensions/MatchCollectionExtensions.netstandard.cs

```diff
@@ -0,0 +1,23 @@
+﻿#if NETSTANDARD
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace System.Text.RegularExpressions
+{
+    public static class MatchCollectionExtensions
+    {
+        public static IEnumerable<Match> Where(this MatchCollection matchCollection, Func<Match,bool> predicate)

```

Comment:
Is it usual for a library to add its own extension methods to standard library types?

Reply:
Let's not make this public. You can rewrite the only usage of this method in `ParseDecimalValueFromOverflowException` as:

```C#
            var matches = regex.Matches(oex.Message);
            foreach (Match match in matches)
            {
                string value = match.Value;
                if (!string.IsNullOrEmpty(value))
                {
                    return value;
                }
            }

            throw oex;
```

Reply:
removed